ako-infra/ingestion/avi_infra_controller.go

```diff
@@ -87,40 +87,40 @@ func (a *AviControllerInfra) VerifyAviControllerLicense() error {
 	return fmt.Errorf("Avi Controller license is not in accepted list %s. License tier is: %s", acceptedLicensesInAvi, *response.DefaultLicenseTier)
 }
 
-func (a *AviControllerInfra) checkNSAnnotations() (error, string) {
-	cloudName := ""
+func (a *AviControllerInfra) checkNSAnnotations(key string) (string, bool) {
 	nsName := utils.GetAKONamespace()
 	nsObj, err := a.cs.CoreV1().Namespaces().Get(context.TODO(), nsName, metav1.GetOptions{})
 	if err != nil {
-		utils.AviLog.Errorf("Failed to GET the %s namespace details due to the following error :%v", nsName, err.Error())
-		return err, cloudName
+		utils.AviLog.Warnf("Failed to GET the %s namespace details due to the following error :%v", nsName, err.Error())
+		return "", false
 	}
-	cloudName = nsObj.Annotations[lib.WCPCloud]
-	if cloudName != "" {
-		utils.AviLog.Infof("Found cloud %s in ns annotation", cloudName)
+	if value, ok := nsObj.Annotations[key]; ok && value != "" {
+		utils.AviLog.Infof("Found key in NS Annotations, key: %s, value: %s", key, value)
+		return value, true
 	}
-	return nil, cloudName
+	return "", false
 }
 
-func (a *AviControllerInfra) checkVirtualService() (error, string) {
-	cloudName := ""
-	vsName := lib.GetClusterName() + "--kube-system-kube-apiserver-lb-svc"
-	SetAdminTenant := session.SetTenant(lib.GetAdminTenant())
-	defer SetAdminTenant(a.AviRestClient.AviSession)
-	SetTenant := session.SetTenant("*")
-	SetTenant(a.AviRestClient.AviSession)
-	uri := "/api/virtualservice/?include_name&created_by=" + lib.GetAKOUser() + "&name=" + vsName
+func (a *AviControllerInfra) checkVirtualService() (string, error) {
+	uri := "/api/virtualservice?include_name=True&name.contains=kube-system-kube-apiserver-lb-svc&se_group_ref.name=" + lib.GetClusterID()
```

Comment:
can we create a lib function for querying the VS "kube-apiserver-lb-svc" and use it?